src/coin_test/data/dataset_generator.py

```diff
@@ -70,6 +70,7 @@ def generate(
             synthetic_df.index = period_index.copy()
             new_datasets.append(
                 self.DATASET_TYPE(
+                    f"dataset_{i}",
```

Comment:
I think it would be nice if the name included the type of dataset generator being used, e.g. `"returns_sampling_dataset_{i}"`or something.

Reply:
Great call, I'll make this change, maybe string the type of class...